src/plugins/mf_priority.cpp

```diff
@@ -204,6 +209,74 @@ static int validate_cb (flux_plugin_t *p,
                                     "user does not belong to specified bank");
     }
 
+    mj_it = users_mj.find (userid);
+
+    if (bank == NULL) {
+        def_bank_it = users_def_bank.find (userid);
+        mj_inner_it = mj_it->second.find (def_bank_it->second);
+
+        if (flux_jobtap_job_aux_set (p,
+                                     FLUX_JOBTAP_CURRENT_JOB,
+                                     "mf_priority:bank",
+                                     &def_bank_it->second[0],
+                                     NULL) < 0)
+            flux_log_error (h, "flux_jobtap_job_aux_set");
+    } else {
+        mj_inner_it = mj_it->second.find (bank);
+
+        if (flux_jobtap_job_aux_set (p,
+                                     FLUX_JOBTAP_CURRENT_JOB,
+                                     "mf_priority:bank",
+                                     &bank[0],
+                                     NULL) < 0)
+            flux_log_error (h, "flux_jobtap_job_aux_set");
+    }
+
+    if (mj_inner_it->second[1] >= mj_inner_it->second[0])
+        return flux_jobtap_reject_job (p, args,
+                                    "user has max number of jobs submitted");
+
+    // increment user's max_jobs count
+    mj_inner_it->second[1]++;
+
+    return 0;
+}
+
+
+static int inactive_cb (flux_plugin_t *p,
+                        const char *topic,
+                        flux_plugin_arg_t *args,
+                        void *data)
+{
+    int userid;
+    void *bank;
+
+    std::map<int, std::map<std::string, std::vector<int>>>::iterator mj_it;
+    std::map<std::string, std::vector<int>>::iterator mj_inner_it;
+
+    std::map<int, std::string>::iterator def_bank_it;
+
+    flux_t *h = flux_jobtap_get_flux (p);
+    if (flux_plugin_arg_unpack (args,
+                                FLUX_PLUGIN_ARG_IN,
+                                "{s:i}",
+                                "userid", &userid) < 0) {
+        flux_log (h,
+                  LOG_ERR,
+                  "flux_plugin_arg_unpack: %s",
+                  flux_plugin_arg_strerror (args));
+        return -1;
+    }
+
+    bank = flux_jobtap_job_aux_get (p,
+                                    FLUX_JOBTAP_CURRENT_JOB,
+                                    "mf_priority:bank");
+
+    mj_it = users_mj.find (userid);
+    mj_inner_it = mj_it->second.find (static_cast<char *> (bank));
+
+    mj_inner_it->second[1]--;
```

Comment:
This is just a general observation: This code is hard to read. It is difficult to follow what `mj_it` and `mj_inner_it` point to and (not your fault) the `first` and `second` members of the iterator make things even more obtuse. Then the fact that max and current job count is stored as a tuple/array(?) the end result is a little confusing. (Disregard this comment if you still plan to collect user/bank information into a `bank_info` struct because I think that will resolve the issue)

Since this is a map type structure, can you not just index into the map, e.g. `users_mj[userid][bank][1]--`? (Sorry if that is a dumb question) You can see it would be even nicer if `users[userid][bank]` contained a struct bank_info because you could just reference members by name:
```
 users[userid][bank]->current_jobs--;
```

Or, if you stored the address of the bank_info struct as the aux item for the job, just:
```
bankinfo->current_jobs--;
```

Reply:
Yeah, I would agree with you. I took the weekend to take a step back and try and re-evaluate adding the max jobs limit support. I think using a `bank_info` struct would make the most sense, so I just pushed one `fixup` commit that attempts to use it. I don't expect it to be perfect but the commit itself contains a few significant changes. 

- the main proposed change is the use of the map data structure containing the user information - instead of adding one new map for every user information component (fairshare, max jobs, any future limits we decide to add in the future), a new map is created that maps a userid and bank combination to a `bank_info` struct:

```c++
struct bank_info {
    double fairshare;
    int max_jobs;
    int current_jobs;
};
```

I kept the default bank map for now because I wasn't sure how else to handle the default bank case.

- the validate callback now fetches `current_jobs` and `max_jobs` information from the `bank_info` struct of the user/bank combination using the C++ map's [`at ()`](http://www.cplusplus.com/reference/map/map/at/) function. This throws an `out_of_range` exception if the value cannot be found.
- the use of the `bank_info` struct I think also simplifies the syntax of incrementing/decrementing the current job count, which no longer makes use of a map iterator:

```c++
users[userid][bank].current_jobs++;
```
- using the struct also removes the need of using an iterator in the priority calculation as well, as we can fetch the fairshare value in a similar way:

```c++
fshare_factor = users[userid][bank].fairshare;
```

Let me know if these changes make sense and help improve the readability at all (I hope they do).

Reply:
This sounds great! I'll try to go through another review and testing today.